synse/http_integration_test.go

```diff
@@ -0,0 +1,551 @@
+package synse
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/vapor-ware/synse-client-go/synse/scheme"
+)
+
+func TestIntegration_Status(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	status, err := client.Status()
+	assert.NoError(t, err)
+	assert.Equal(t, "ok", status.Status)
+	assert.NotEmpty(t, status.Timestamp)
+}
+
+func TestIntegration_Version(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	version, err := client.Version()
+	assert.NoError(t, err)
+	assert.Equal(t, "v3", version.APIVersion)
+	assert.NotEmpty(t, version.Version)
+}
+
+func TestIntegration_Config(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	config, err := client.Config()
+	assert.NoError(t, err)
+	assert.NotEmpty(t, config)
+}
+
+func TestIntegration_Plugins(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	plugins, err := client.Plugins()
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(plugins))
+
+	for _, plugin := range plugins {
+		assert.NotEmpty(t, plugin.Name)
+		assert.NotEmpty(t, plugin.Maintainer)
+		assert.NotEmpty(t, plugin.Tag)
+		assert.NotEmpty(t, plugin.Description)
+		assert.NotEmpty(t, plugin.ID)
+		assert.True(t, plugin.Active)
+	}
```

Comment:
I think that since we control the integration test environment and know that there should only ever be the one emulator plugin, instead of iterating through the slice and doing a generic check, we could do checks specific to the emulator, e.g.

```go
p := plugins[0]
assert.Equal(t, "emulator", plugin.Name)
assert.Equal(t, "vaporio", plugin.Maintainer)
...
```